subsys/debug/thread_awareness.c

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright (c) 2017 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <kernel.h>
+
+/* ZTA stands for Zephyr Thread Awareness */
+#define ZTA_UNIMPLEMENTED	0xffffffff
+
+#if defined(CONFIG_DEBUG_THREAD_INFO) && defined(CONFIG_THREAD_MONITOR)
+enum {
+	ZTA_OFFSET_VERSION,
+	ZTA_OFFSET_K_CURR_THREAD,
+	ZTA_OFFSET_K_THREADS,
+	ZTA_OFFSET_T_ENTRY,
+	ZTA_OFFSET_T_NEXT_THREAD,
+	ZTA_OFFSET_T_STATE,
+	ZTA_OFFSET_T_USER_OPTIONS,
+	ZTA_OFFSET_T_PRIO,
+	ZTA_OFFSET_T_STACK_PTR,
+	ZTA_OFFSET_T_NAME,
+	ZTA_OFFSET_T_ARCH,
+	ZTA_OFFSET_T_PREEMPT_FLOAT,
+	ZTA_OFFSET_T_COOP_FLOAT,
+};
+
+#if CONFIG_MP_NUM_CPUS > 1
+#error "This code doesn't work properly with multiple CPUs enabled"
+#endif
+
+/* Forward-compatibility notes: 1) Only append items to this table; otherwise
+ * debugger plugin versions that expect fewer items will read garbage values.
+ * 2) Avoid incompatible changes that affect the interpretation of existing
+ * items. But if you have to do them, increment ZTA_OFFSET_VERSION
+ * and submit a patch for debugger plugins to deal with both the old and new
+ * scheme.
+ * Only version 1 is backward compatible to version 0.
+ */
+__attribute__((used, section(".zta")))
+size_t _kernel_zta_offsets[] = {
```

Comment:
Those symbols are used outside Zephyr project, and cannot simply be renamed without ensuring dependents are updated as well.
- `_kernel_openocd_offsets`
- `_kernel_openocd_size_t_size`

From [pyOCD](https://github.com/pyocd/pyOCD)

https://github.com/pyocd/pyOCD/blob/ade9b685a41d22176c6e85c3c70c2d343dcf73d6/pyocd/rtos/zephyr.py#L247-L251
```
    ZEPHYR_SYMBOLS = [
        "_kernel",
        "_kernel_openocd_offsets",
        "_kernel_openocd_size_t_size",
        ]
```

Reply:
I see the aliases now :+1: